zcash_client_sqlite/src/wallet/db.rs

```diff
@@ -41,18 +41,16 @@ CREATE TABLE "accounts" (
     recover_until_height INTEGER,
     has_spend_key INTEGER NOT NULL DEFAULT 1,
     CHECK (
-        (
+      (
         account_kind = 0
         AND hd_seed_fingerprint IS NOT NULL
         AND hd_account_index IS NOT NULL
         AND ufvk IS NOT NULL
-        )
-        OR
-        (
+      )
+      OR
+      (
         account_kind = 1
-        AND hd_seed_fingerprint IS NULL
-        AND hd_account_index IS NULL
```

Comment:
This is the removed constraint that now permits data corruption to go undetected (because we aren't requiring the columns to be either both null, or both not null).